src/saltext/github/states/github.py

```diff
@@ -743,3 +743,175 @@ def repo_absent(name, profile="github", **kwargs):
             ] = f"Failed to delete repo {name}. Ensure the delete_repo scope is enabled if using OAuth."
             ret["result"] = False
     return ret
+
+
+def ruleset_absent(
+    name, ruleset_type, profile="github", owner=None, repo_name=None, org_name=None, **kwargs
+):
+    """
+    Ensure a ruleset is absent.
+
+    Example:
+
+    .. code-block:: yaml
+
+        ensure rulset test is absent in github:
+            github.ruleset_absent:
+                - name: 'test'
+                - ruleset_type: 'repo'
+                - owner: test
+                - repo_name: test_repo
+
+    The following parameters are required:
+
+    name
+        name of the ruleset.
+    ruleset_type
+        type of ruleset ('org' or 'repo')
+    """
+
+    ret = {"name": name, "changes": {}, "result": None, "comment": ""}
+
+    if ruleset_type not in ["org", "repo"]:
+        raise CommandExecutionError
```

Comment:
**issue (blocking):** in state module functions, we don't raise exceptions. we just return the `ret` dictionary with `result: False` to show failure and set the `comment` to something that tells us why.